pymc3/distributions/distribution.py

```diff
@@ -178,12 +178,21 @@ def __init__(self, shape=(), dtype=None, defaults=('median', 'mean', 'mode'),
 class DensityDist(Distribution):
     """Distribution based on a given log density function."""
 
-    def __init__(self, logp, shape=(), dtype=None, testval=0, *args, **kwargs):
+    def __init__(self, logp, random=None, shape=(), dtype=None, testval=0, *args, **kwargs):
         if dtype is None:
             dtype = theano.config.floatX
         super(DensityDist, self).__init__(
             shape, dtype, testval, *args, **kwargs)
         self.logp = logp
+        self.rand = random
+    
+    def random(self):
+        if self.rand is not None:
+            return self.rand
```

Comment:
I don't think this works, you need to call `self.rand(*args, **kwargs)` and add those as well to the method call signature.

Reply:
Yeah that makes more sense. :+1: